tests/e2e/tests/critical-path/tree-view/tree-view-improvements.e2e copy.ts

```diff
@@ -0,0 +1,140 @@
+            `HSET ${keyNames[4]} field value`,
+        ];
+
+        // Create 4 keys and index
+        await cliPage.sendCommandsInCli(commands1);
+        await t.click(browserPage.refreshKeysButton); // refresh keys
+
+        await t.expect(
+            treeItemKeys.visible).
+            ok("Folder is not selected");
+        await verifyKeysDisplayedInTheList([firstTreeItemText + ":1", firstTreeItemText + ":2"]); // verify created keys are visible
+
+        await browserPage.selectFilterGroupType(KeyTypesTexts.Hash);
+
+        await t.expect(treeItemKeys.visible).ok("Folder is not selected after searching with HASH");
+        await verifyKeysDisplayedInTheList([firstTreeItemText + ":1", firstTreeItemText + ":2"]); // verify created keys are visible
+
+        await browserPage.searchByKeyName('*');
+
+        await t.expect(treeItemKeys.visible).ok("Folder is not selected");
+        await verifyKeysDisplayedInTheList([firstTreeItemText + ":1", firstTreeItemText + ":2"]); // verify created keys are visible
+
+        await t.click(browserPage.clearFilterButton);
+
+        await t.expect(treeItemKeys.visible).ok("Folder is not selected");
+        await verifyKeysDisplayedInTheList([firstTreeItemText + ":1", firstTreeItemText + ":2"]); // verify created keys are visible
+
+        await browserPage.selectFilterGroupType(KeyTypesTexts.Stream);
+
+        await t.expect(
+            Selector(`[role="rowgroup"]`).find(`div`).withText("No results found.").visible).
+            ok("No results text is visible"); // verify no results found
+
+        await t.click(browserPage.streamDeleteButton); // clear stream from filter
+
+        await t.expect(
+            Selector(`[role="rowgroup"]`).find(`div`).withText("No results found.").visible).
+            notOk("No result text is still visible");
+        await t.expect(
+            treeItemKeys.visible).
+            notOk("First folder is expanded");
+
+    });
+
+test
+    .before(async () => {
+        await acceptLicenseTermsAndAddDatabaseApi(ossStandaloneBigConfig, ossStandaloneBigConfig.databaseName);
+    })
+    .after(async () => {
+        await cliPage.sendCommandInCli(`FT.DROPINDEX ${index}`);
+        await deleteStandaloneDatabaseApi(ossStandaloneBigConfig);
+    })
+    ('Verify tree view navigation for index based search', async t => {
+        // generate index based on keyName
+        let folders = ["mobile", "2"];
+
+        index = await cliPage.createRandomIndexNamewithCLI();
+
+        await t.click(browserPage.redisearchModeBtn); // click redisearch button
+        await browserPage.selectIndexByName(index);
+
+        await t.click(browserPage.treeViewButton);
+        await browserPage.openTreeFolders(folders);
+        await t.click(browserPage.refreshKeysButton);
+
+        await t.expect(
+            Selector(`[data-testid="node-item_${folders[0]}:${folders[1]}:keys:keys:"]`).visible)
+            .ok("Folder is not selected");
+
+    });
```

Comment:
Don't see these tests:
1) "The folder without any patterns selected and the list of keys is displayed when there is a folder without any patterns"
![image](https://user-images.githubusercontent.com/90257586/206441449-b6cf7faf-9b2a-4a03-9949-5421eb6ee691.png)
2) The folder without any namespaces is selected (if exists) when folder does not exist after search/filter
![image](https://user-images.githubusercontent.com/90257586/206441846-d49a944a-813c-40d1-95a0-c35b81fb71d7.png)
3) The first folder with namespaces is expanded and selected when folder and folder without any namespaces does not exist after search/filter
(it is when one folder is opened and you search for key from other folder - other folder should be opened after such search)
4) The previously selected folder is preselected when key does not exist after keys refresh (you need to select one folder, then delete any key from this folder by CLI and click refresh - verify that the same folder still selected)